python/lsst/obs/base/gen2to3/repoWalker/scanner.py

```diff
@@ -0,0 +1,204 @@
+# This file is part of obs_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ["PathElementHandler", "DirectoryScanner"]
+
+from abc import ABC, abstractmethod
+import bisect
+import os
+from typing import (
+    Callable,
+    Iterator,
+    List,
+    Mapping,
+    Optional,
+)
+
+from lsst.log import Log
+from lsst.daf.butler import (
+    DataCoordinate,
+    DatasetType,
+    FileDataset,
+)
+
+
+class PathElementHandler(ABC):
```

Comment:
Would it make sense to move this class to the `handlers.py` file?

Reply:
It needs to be here to keep the dependencies between modules non-circular.  I'll add a common or module docstring to both modules explaining this.